### PR TITLE
Make imports dynamic based on command for faster startup

### PR DESCRIPTION
**Pull request by sbgillett**
At present `__main__.py` imports all modules each time the `pymobiledevice3` script is invoked. This can be slow on some systems and makes working programmatically with pymobiledevice3 undesirable at scale.

This change makes Click module loading more dynamic and thus improves performance; I'm seeing speed-up of around 1.4 seconds on my test bench (EC2 `t2.xlarge` running AL2 and Python 3.9.16) when averaging all subcommand invocations.

```
pymobiledevice3 activation: Now - 0.62 seconds | Was - 2.05 seconds | Difference - 1.43 seconds
pymobiledevice3 afc: Now - 0.68 seconds | Was - 2.04 seconds | Difference - 1.36 seconds
pymobiledevice3 amfi: Now - 0.61 seconds | Was - 2.05 seconds | Difference - 1.44 seconds
pymobiledevice3 apps: Now - 0.66 seconds | Was - 2.03 seconds | Difference - 1.37 seconds
pymobiledevice3 backup2: Now - 0.66 seconds | Was - 2.05 seconds | Difference - 1.39 seconds
pymobiledevice3 bonjour: Now - 0.62 seconds | Was - 2.06 seconds | Difference - 1.44 seconds
pymobiledevice3 companion: Now - 0.60 seconds | Was - 2.05 seconds | Difference - 1.45 seconds
pymobiledevice3 crash: Now - 0.67 seconds | Was - 2.02 seconds | Difference - 1.35 seconds
pymobiledevice3 developer: Now - 0.92 seconds | Was - 2.02 seconds | Difference - 1.10 seconds
pymobiledevice3 diagnostics: Now - 0.62 seconds | Was - 2.07 seconds | Difference - 1.45 seconds
pymobiledevice3 lockdown: Now - 0.63 seconds | Was - 2.08 seconds | Difference - 1.45 seconds
pymobiledevice3 mounter: Now - 0.71 seconds | Was - 2.05 seconds | Difference - 1.34 seconds
pymobiledevice3 notification: Now - 0.61 seconds | Was - 2.03 seconds | Difference - 1.42 seconds
pymobiledevice3 pcap: Now - 0.62 seconds | Was - 2.09 seconds | Difference - 1.47 seconds
pymobiledevice3 power-assertion: Now - 0.62 seconds | Was - 2.04 seconds | Difference - 1.42 seconds
pymobiledevice3 processes: Now - 0.63 seconds | Was - 2.06 seconds | Difference - 1.43 seconds
pymobiledevice3 profile: Now - 0.62 seconds | Was - 2.04 seconds | Difference - 1.42 seconds
pymobiledevice3 provision: Now - 0.61 seconds | Was - 2.11 seconds | Difference - 1.50 seconds
pymobiledevice3 remote: Now - 1.36 seconds | Was - 2.03 seconds | Difference - .67 seconds
pymobiledevice3 restore: Now - 0.75 seconds | Was - 2.08 seconds | Difference - 1.33 seconds
pymobiledevice3 springboard: Now - 0.62 seconds | Was - 2.05 seconds | Difference - 1.43 seconds
pymobiledevice3 syslog: Now - 0.61 seconds | Was - 2.02 seconds | Difference - 1.41 seconds
pymobiledevice3 usbmux: Now - 0.60 seconds | Was - 2.03 seconds | Difference - 1.43 seconds
pymobiledevice3 webinspector: Now - 1.35 seconds | Was - 2.04 seconds | Difference - .69 seconds
```

### CODE REVIEW

**Comment by doronz88** [comment]
Really liked the idea! Please add a `if __name__ == '__main__'`  so the module can be invoked using `python3 -m pymobiledevice3`

**Comment by sbgillett** [comment]
> Really liked the idea! Please add a `if __name__ == '__main__'` so the module can be invoked using `python3 -m pymobiledevice3`

Done! 👍

File: pymobiledevice3/__main__.py [hunk 1]
```diff
@@ -64,62 +31,58 @@
...
 
 
+class Pmd3Cli(click.Group):
+    def list_commands(self, ctx):
+        return CLI_GROUPS.keys()
+
+    def get_command(self, ctx, name):
+        try:
+            mod = __import__(f"pymobiledevice3.cli.{CLI_GROUPS[name]}", None, None, ['cli'])
+        except ImportError:
```
Comment (doronz88) [inline pymobiledevice3/__main__.py:83]: In order to preserve old behavior please raise the proper click error.
For example when I run:

```shell
pymobiledevice3 non-existing-subcommand
```

I'm expecting:
```
Usage: pymobiledevice3 [OPTIONS] COMMAND [ARGS]...
Try 'pymobiledevice3 -h' for help.

Error: No such command 'non-existing-subcommand'.
```

**Review by doronz88 — CHANGES_REQUESTED** [review]

**Review by doronz88 — CHANGES_REQUESTED** [review]
Also, please your changes into a single commit:
```
cli: use dynamic imports to improve performance
```

File: pymobiledevice3/__main__.py (hunk 1 above)
Comment (sbgillett) [reply pymobiledevice3/__main__.py:83]: Fixed with [8dcbc58](https://github.com/doronz88/pymobiledevice3/pull/702/commits/8dcbc5840832e170fa062fb119d514c4fd32e57b)

**Comment by sbgillett** [comment]
> `cli: use dynamic imports to improve performance`

You should be able to [squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits). Alternatively I can cancel this PR and create a new one - let me know.

**Review by doronz88 — APPROVED** [review]
In general I'm very much against squashing and losing the history. I use the commit messages to produce the changelog.

Anyhow, I'll just take care for the commits (no need to open another PR for such stuff. just a matter of a simple `git rebase`)

**Pull request merged**